mobius/observability/__init__.py

```
@@ -0,0 +1,292 @@
+
+AuditStorageLike = Union[AuditStorage, Path, str]
+
+
+def _resolve_storage(storage: Optional[AuditStorageLike], audit_path: Optional[Union[str, Path]]) -> AuditStorage:
+    if storage is None:
+        return build_default_storage(audit_path)
+    if isinstance(storage, (str, Path)):
+        return JSONLStorage(storage)
+    if hasattr(storage, "append"):
+        return storage  # type: ignore[return-value]
+    raise TypeError("Unsupported audit storage type provided")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)